### ISSUE TITLE
Id member not required when resource object represents a new resource to be created on the server

### ISSUE THREAD

**Issue by dynamiccast** [elliotttf/jsonapi-validator#29]
Hi,

_jsonapi-validator_ requires resources to have an `id` https://github.com/elliotttf/jsonapi-validator/blob/master/lib/schema.json#L122

However, there is an exception to this rule in the JSON API specification (http://jsonapi.org/format/#document-resource-objects).

> Exception: The id member is not required when the resource object originates at the client and represents a new resource to be created on the server.

Should _jsonapi-validator_ be updated to take this case into account ?

**Comment by elliotttf** [comment]
Yes! This is an issue that should be fixed in [jsonapi](https://github.com/json-api/json-api) though. You can probably just take that schema and apply it directly.
